### DISCUSSION TITLE
Another issue related to the relationship query.  the relationship levels issues.

### DISCUSSION THREAD

**Discussion by smartbox-jiandonghan** [rebing/graphql-laravel#1086] [Q&A]
I have 3 levels of 'one to many' relationships.
level1. Agent,
level2. Customer, one agent can have many  customers
level 3.Transction, one customer can have many  transactions

for the relationship query, I declared the agent model, customer model and transaction model, then declared the 'hasMany' and 'belongsTo' relationship inside.
then I use the following code to excecute the relationship query.
```
    $fields = $getSelectFields();
    $select = $fields->getSelect();
    $with = $fields->getRelations();

    $agents = Agent::select($select)->with($with)
    ->where('agent_id', $args['agent_id']);
    
    return $agents->get();  
```

I found, when I used 2 levels relationship query. for example:
```
 agents (agent_id: XXX) {
    agent_id,
    agent_name,
    customers {
  	 customer_id,
  	 customer_name,
         }
   }  
```
it runs ok.

but when I used 3 levels relationship query. for example:
```
 agents (agent_id: XXX) {
     agent_id,
     agent_name,
     customers {
  	 customer_id,
  	 customer_name,
        transactions {
              transaction_date,
              transaction_amount
              }
         }
     }  
```
It doesn't work. no any response returned, but no errors prompted.
I use the QraphiQL to test it. 

Did you ever encountered this issue before?  what reasons may cause this result?

Thanks a lot for any help.

**Comment by mfn** [comment]
What is the actual issue? Wrong result? Do you get an error? Do you have a stacktrace?

**Reply by smartbox-jiandonghan** [reply]
Hello Markus,
when I tested the 3 levels relationship query in GraphiQL, no response returned, but no errors prompted.
and no error messages in laravel.log.

thanks.

**Reply by mfn** [reply]
When you turn on logging, what SQL statement are you seeing? Do you spot anything odd there?

I think the best case for helping is to have a (standalone) reproducible small project using sqlite and some instruction for the setup.

Or: you make a PR with a test to this repo doing this kind of queries demonstrating how it fails (sqlite, models, etc. can be created) and we can analyze this.

**Reply by smartbox-jiandonghan** [reply]
Ok. thank you Markus. will do it.
